stac_generator/raster_data/generator.py

```diff
@@ -29,105 +39,133 @@ def __init__(
+
+                # Ensure datetime is timezone-aware
+                if source_cfg.datetime.tzinfo is None:
+                    datetime_aware = source_cfg.datetime.replace(tzinfo=UTC)
+                else:
+                    datetime_aware = source_cfg.datetime
+
+                # Create STAC Item
                 item = pystac.Item(
                     id=source_cfg.prefix,
-                    geometry=geometry,
+                    geometry=geometry_geojson,
                     bbox=bbox,
-                    datetime=source_cfg.datetime,
-                    properties={}  # properties will be managed by extensions
+                    datetime=datetime_aware,
+                    properties={
+                        "eo:snow_cover": 0,
+                        "eo:cloud_cover": 0,
+                        "proj:epsg": crs.to_epsg(),
+                        "proj:shape": shape,
+                        "proj:transform": list(transform)[:9],
+                    },
                 )
 
-                # Initialize EOExtension and set eo:bands as Band objects
-                eo_ext = EOExtension.ext(item, add_if_missing=True)
+                # Initialize extensions
+                EOExtension.ext(item, add_if_missing=True)
+                proj_ext = ProjectionExtension.ext(item, add_if_missing=True)
+                proj_ext.apply(epsg=crs.to_epsg(), shape=shape, transform=list(transform)[:9])
+
+                # Create EO and Raster bands
                 eo_bands = []
+                raster_bands = []
                 for idx, band_info in enumerate(source_cfg.bands, start=1):
-                    band = Band.create(
+                    eo_band = Band.create(
                         name=f"B{idx}",
                         common_name=BAND_MAPPING.get(band_info.name.lower(), None),
-                        center_wavelength=float(band_info.wavelength) if band_info.wavelength not in [None, "no band specified"] else None
+                        center_wavelength=band_info.wavelength,
+                        description=f"Common name: {BAND_MAPPING.get(band_info.name.lower(), 'unknown')}",
                     )
-                    eo_bands.append(band)
+                    eo_bands.append(eo_band)
 
-                # Ensure eo:bands is managed by EOExtension only
-                eo_ext.bands = eo_bands
+                    raster_band = RasterBand.create(
+                        nodata=band_info.nodata or 0,
+                        data_type=DataType(band_info.data_type or "uint16"),
+                    )
+                    raster_bands.append(raster_band)
 
-                # Update other properties
-                item.properties.update({
-                    "gsd": float(src.res[0]),
-                    "platform": "UAV",
-                    "instruments": ["Multispectral Camera"],
-                })
+                item.properties["eo:bands"] = [band.to_dict() for band in eo_bands]
```

Comment:
Great work getting the band attributes from the config file. On line 152 and 153 you add the raster extension fields using the apply method on the extension but here and on line 156 you are manually adding to keys in a dictionary. I prefer the method you used for the raster extension.

Reply:
Use .apply()

Reply:
done